v2/pkg/errorcodes/const.go

```diff
@@ -1,8 +1,9 @@
 package errorcodes
 
 const (
-	BadUserInput            = "BAD_USER_INPUT"
-	GraphQLValidationFailed = "GRAPHQL_VALIDATION_FAILED"
-	InternalServerError     = "INTERNAL_SERVER_ERROR"
-	InvalidGraphql          = "INVALID_GRAPHQL"
+	BadUserInput                  = "BAD_USER_INPUT"
+	GraphQLValidationFailed       = "GRAPHQL_VALIDATION_FAILED"
+	InternalServerError           = "INTERNAL_SERVER_ERROR"
+	InvalidGraphql                = "INVALID_GRAPHQL"
+	ValidationInvalidTypeVariable = "VALIDATION_INVALID_TYPE_VARIABLE"
```

Comment:
what about splitting into 2 const items

and put a comments `// gateway compatibility`

and `// router compatibility`

Reply:
Are we sure this is not a valid error code in gateway?

Reply:
https://sourcegraph.com/search?q=context:global+repo:apollographql/federation+VALIDATION_INVALID_TYPE_VARIABLE&patternType=keyword&sm=0

suggests it is not

Reply:
Done